### PR TITLE
*: merge dev to master

### PR DESCRIPTION
**Pull request by disksing**
Code changes are identical to #1312 

@siddontang

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @ngaut

**Comment by siddontang** [comment]
We will remove `dev` after merged.

File: src/storage/mvcc/txn.rs [hunk 1]
```diff
@@ -20,10 +20,13 @@ use super::write::{WriteType, Write};
 use super::{Error, Result};
 use super::metrics::*;
 
+pub const MAX_TXN_WRITE_SIZE: usize = 512 * 1024;
```
Comment (ngaut) [inline src/storage/mvcc/txn.rs]: s/512/32/

**Comment by ngaut** [comment]
LGTM

**Pull request merged**